src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource.cs

```diff
@@ -277,24 +273,39 @@ void Disconnect()
                 }
             }
 
-            private void Pause()
+            private bool IsVisible()
+                => _visibilityTracker == null || _visibilityTracker.IsVisible(_subjectBuffer);
+
+            private void PauseIfNotVisible()
             {
                 _dataSource.ThreadingContext.ThrowIfNotOnUIThread();
-                _paused = true;
-                _eventSource.Pause();
+
+                if (!IsVisible())
+                {
+                    _paused = true;
+                    _eventSource.Pause();
+                }
             }
 
-            private void Resume()
+            private void ResumeIfVisible()
             {
                 _dataSource.ThreadingContext.ThrowIfNotOnUIThread();
+
                 // if we're not actually paused, no need to do anything.
-                if (_paused)
-                {
-                    // Set us back to running, and kick off work to compute tags now that we're visible again.
-                    _paused = false;
-                    _eventSource.Resume();
-                    EnqueueWork(initialTags: false);
-                }
+                if (!_paused)
+                    return;
+
+                // If we're not visible, no need to resume.
+                if (!IsVisible())
+                    return;
+
+                // Set us back to running, and kick off work to compute tags now that we're visible again.
+                _paused = false;
+                _eventSource.Resume();
+
+                // We just transitioned to being visible, compute our tags at high priority so the view is updated as
+                // quickly as possible.
+                EnqueueWork(highPriority: true);
```

Comment:
changed this to high-pri as i noticed a delay on some tagging when switching between files.